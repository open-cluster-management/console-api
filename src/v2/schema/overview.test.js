/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Overview Resolver', () => {
  test('Correctly Resolves Overview Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
        overview {
          clusters {
            metadata {
              name
              namespace
              labels
              uid
            }
            capacity {
              cpu
              memory
            }
            allocatable {
              cpu
              memory
            }
            consoleURL
            status
          }
          applications {
            metadata {
              name
              namespace
            }
            raw
            selector
          }
          compliances {
            metadata {
              name
              namespace
            }
            raw
          }
        }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
