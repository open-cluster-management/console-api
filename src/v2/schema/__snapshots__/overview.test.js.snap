// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Resolver Correctly Resolves Overview Query 1`] = `
Object {
  "data": Object {
    "overview": Object {
      "applications": Array [
        Object {
          "metadata": Object {
            "name": "gbapp-gbapp",
            "namespace": null,
          },
          "raw": null,
          "selector": null,
        },
      ],
      "clusters": Array [
        Object {
          "allocatable": Object {
            "cpu": "6598m",
            "memory": "6728Mi",
          },
          "capacity": Object {
            "cpu": "18",
            "memory": "32013Mi",
          },
          "consoleURL": null,
          "metadata": Object {
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "toronto",
              "environment": "Dev",
              "name": "managed-cluster",
              "owner": "marketing",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "managed-cluster",
            "namespace": "managed-cluster",
            "uid": null,
          },
          "status": "pendingimport",
        },
        Object {
          "allocatable": Object {
            "cpu": "6598m",
            "memory": "6728Mi",
          },
          "capacity": Object {
            "cpu": "16",
            "memory": "32013Mi",
          },
          "consoleURL": null,
          "metadata": Object {
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "raleigh",
              "environment": "Dev",
              "name": "hub-cluster",
              "owner": "development",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "hub-cluster",
            "namespace": "hub-cluster",
            "uid": null,
          },
          "status": "ok",
        },
        Object {
          "allocatable": Object {
            "cpu": "7483m",
            "memory": "5390Mi",
          },
          "capacity": Object {
            "cpu": "16",
            "memory": "32013Mi",
          },
          "consoleURL": null,
          "metadata": Object {
            "labels": Object {
              "cloud": "IBM",
              "datacenter": "raleigh",
              "environment": "Dev",
              "name": "new-cluster",
              "owner": "development",
              "region": "US",
              "vendor": "ICP",
            },
            "name": "new-cluster",
            "namespace": "new-cluster",
            "uid": null,
          },
          "status": "offline",
        },
      ],
      "compliances": Array [
        Object {
          "metadata": null,
          "raw": Object {
            "status": Object {},
          },
        },
        Object {
          "metadata": null,
          "raw": Object {
            "status": Object {},
          },
        },
      ],
    },
  },
}
`;
